packages/financial-templates-lib/src/helpers/multicall.js

```diff
@@ -0,0 +1,75 @@
+// Provides convenience methods for interacting with deployed Multicall contract on network.
+const { getAbi } = require("@uma/core");
+const multicallAddressMap = require("@makerdao/multicall/src/addresses.json");
+const assert = require("assert");
+
+// Simulate submitting a batch of `transactions` to the multicall contact
+// and return an array of simulated output values. Caller will need to
+// decode the return values via web3.eth.abi.decodeParameters([types], outputBytes).
+const aggregateTransactionsAndCall = async (multicallAddress, web3, transactions) => {
+  const multicallContract = new web3.eth.Contract(getAbi("Multicall"), multicallAddress);
+  for (let i = 0; i < transactions.length; i++) {
+    assert(
+      transactions[i].target && transactions[i].callData,
+      "transaction expected in form {target: address, callData: bytes}"
+    );
+  }
+  return await multicallContract.methods.aggregate(transactions).call();
+};
+
+// Dictionary of ABI's for contract methods that we can use to decode
+// to Javascript type.
+const contractMethodABIs = {
+  fundingRate: [
```

Comment:
is there no way to extract this directly from the contract being called? why does it need to be hard coded like this? this will limit how extendable this helper is. could you not find this same ABI directly from the truffle artefact?

Reply:
Good point, replaced with `this.financialContractAbi.find(method => method.name === "fundingRate").outputs`